pkg/controllers/hpareplicassyncer/hpa_replicas_syncer_controller.go

```diff
@@ -125,8 +126,14 @@ func (r *HPAReplicasSyncer) updateScaleIfNeed(ctx context.Context, workloadGR sc
 
 		return nil
 	}
-
-	if scale.Spec.Replicas != hpa.Status.DesiredReplicas {
+	// DesiredReplicas only has meaning when ScalingActive and AbleToScale are both True. At any other time, it is meaningless.
+	// So the workload does not need to be updated.
+	// Refer to:
+	// https://github.com/kubernetes/kubernetes/blob/v1.28.0/pkg/controller/podautoscaler/horizontal.go#L797
+	// https://github.com/kubernetes/kubernetes/blob/v1.28.0/pkg/controller/podautoscaler/horizontal.go#L343
+	if scale.Spec.Replicas != hpa.Status.DesiredReplicas &&
+		findConditionTrue(hpa.Status, autoscalingv2.ScalingActive) &&
```

Comment:
Would it be better to do the judgement of conditions earlier? Such as after got HPA in `Reconcile`, so that we could reduce a lot of operations.